megamek/src/megamek/common/Aero.java

```diff
@@ -2152,13 +2152,20 @@ public int calculateBattleValue(boolean ignoreC3, boolean ignorePilot) {
 
             // sort the heat-using weapons by modified BV
             Collections.sort(heatBVs, (obj1, obj2) -> {
+                Double obj1BV = (Double) obj1.get(0); // BV
+                Double obj2BV = (Double) obj2.get(0); // BV
+                
                 // first element in the the ArrayList is BV, second is heat
                 // if same BV, lower heat first
-                if (obj1.get(0).equals(obj2.get(0))) {
-                    return (int) Math.ceil((Double) obj1.get(1) - (Double) obj2.get(1));
+                if(obj1BV == obj2BV) {
```

Comment:
Is reference equality really intended here?

Reply:
Nah, should probably be value equality.